src/main/java/dialog/TaskDialog.java

```diff
@@ -0,0 +1,193 @@
+            archive.put(id, newDialog);
+            return newDialog;
+        }
+    }
+
+    /**
+     * Factory method for creating TaskDialog as Dialog type (to be cast by developer)
+     * Factory method from TaskDialog allow the user to fetch the existing Dialog in the archive.
+     *
+     * @param id       the id of the TaskDialog
+     * @param taskList the taskList the TaskDialog is going to have
+     * @return TaskDialog of type Dialog
+     */
+    public static Dialog generate(String id, TaskList taskList) {
+        if (archive.containsKey(id)) {
+            return archive.get(id);
+        } else {
+            final Dialog newDialog = new TaskDialog(new ArrayList<>(), taskList);
+            archive.put(id, newDialog);
+            return newDialog;
+        }
+    }
+
+    /**
+     * Add the task to the TaskList of this Dialog and print out feedback to the user
+     *
+     * @param task the task to be added to the TaskList of this TaskDialog
+     * @throws DialogException dialog cannot have the same id while the app is running
+     */
+    public void addTask(Task task) throws DialogException {
+        Dialog addDialog = Dialog.generate(Integer.toString(task.hashCode()));
+        taskList.add(task);
+        addDialog.add("Got it. I've added this task:");
+        addDialog.add("  " + task);
+        addDialog.add("Now you have " + this.taskList.length() + " tasks in the list.");
+        System.out.println(addDialog);
+    }
+
+    /**
+     * Return TaskDialog filtered by the specific deadline of format yyyy-MM-dd
+     *
+     * @param deadline the deadline of format yyyy-MM-dd
+     * @return TaskDialog with new TaskList stored in it
+     * @throws InvalidTimeFormatException time of illegal format (not yyyy-MM-dd)
+     */
+    public TaskDialog by(String deadline) throws InvalidTimeFormatException {
+        LocalDate dlDate = Parser.parseTimeString(deadline);
+        return new TaskDialog(new ArrayList<>(List.of("task.Deadline: "
+                + dlDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")))),
+                new TaskList(new ArrayList<>(this.taskList.getTasks().stream().filter((task) -> {
+                    if (!task.getClass().getSuperclass().getName().equals("task.TimeTask")) {
+                        // return any task without time associated with it
+                        return true;
+                    } else {
+                        TimeTask timetask = (TimeTask) task;
+                        return timetask.getTime().isBefore(dlDate) || timetask.getTime().isEqual(dlDate);
+                    }
+                }).collect(Collectors.toList()))));
+    }
+
+    /**
+     * Return TaskDialog filtered by the keyword given by the user
+     *
+     * @param keyword the keyword to search description
+     * @return TaskDialog with TaskList with tasks matching the keyword
+     */
+    public TaskDialog with(String keyword) {
```

Comment:
I would suggest renaming this method to have a verb for a name too, perhaps findFromKeyword.

Reply:
Same as previous comment